File: CS-Examples/02_Data/ExportDataKeepDataFormat.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace ExportDataKeepDataFormat
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Instantiate a new workbook object.
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ExportDataKeepDataFormat.xlsx")

            ' Access the first worksheet in the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Instantiate a new ExportTableOptions object to specify export options.
            Dim options As New ExportTableOptions()

            ' Set the option to not keep the data format during export.
            options.KeepDataFormat = False

            'Set the option to rename strategy as "Digit" during export.
            options.RenameStrategy = RenameStrategy.Digit

            ' Export the data from the specified range of the worksheet to a DataTable object.
            Dim table As DataTable = sheet.ExportDataTable(1, 1, sheet.LastDataRow, sheet.LastDataColumn, options)

            ' Set the DataGridView's data source as the exported DataTable.
            Me.dataGridView1.DataSource = table
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/02_Data/ExportDataKeepDataFormat.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ExportDataKeepDataFormat
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
			// Create a workbook
			Workbook workbook = new Workbook();

			// Load the file from disk
			workbook.LoadFromFile(@"../../../../../../Data/ExportDataKeepDataFormat.xlsx");
			
			// Get the first worksheet
			Worksheet sheet = workbook.Worksheets[0];
			
			// Export DataTable without keeping data format
			ExportTableOptions options = new ExportTableOptions();
			options.KeepDataFormat = false;
			options.RenameStrategy = RenameStrategy.Digit;

			// Export data to data table
			DataTable table = sheet.ExportDataTable(1, 1, sheet.LastDataRow, sheet.LastDataColumn, options); 
			
			// Show the data table
            this.dataGridView1.DataSource = table;
			
			// Dispose of the workbook object to free up resources
			workbook.Dispose();
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
